store/store_options.go

```diff
@@ -7,27 +7,18 @@ import (
 type Parameters struct {
 	// RecentBlocksCacheSize is the size of the cache for recent blocks.
 	RecentBlocksCacheSize int
-
-	// AvailabilityCacheSize is the size of the cache for accessors requested for serving availability
-	// samples.
-	AvailabilityCacheSize int
 }
 
 // DefaultParameters returns the default configuration values for the EDS store parameters.
 func DefaultParameters() *Parameters {
 	return &Parameters{
 		RecentBlocksCacheSize: 10,
-		AvailabilityCacheSize: 128,
 	}
 }
 
 func (p *Parameters) Validate() error {
-	if p.RecentBlocksCacheSize < 1 {
-		return errors.New("recent eds cache size must be positive")
-	}
-
-	if p.AvailabilityCacheSize < 1 {
-		return errors.New("availability cache size must be positive")
+	if p.RecentBlocksCacheSize < 0 {
+		return errors.New("recent eds cache size cannot be negative")
 	}
```

Comment:
If constructor now allows disabling cache by taking 0, we should remove this check.

Reply:
Also, I recommend adopting simplification changes I made to tests in https://github.com/celestiaorg/celestia-node/pull/3619/files#diff-ad5b6c129e998456b5ae539dd2a5793f7d094b64ed8175edeeecade80c39e008R339-R344

Reply:
This check prevents negative values of CacheSize to be passed